File: 한민선/240511/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.*;

public class Main {
    public static int getSolution(int startRow, int startCol, String[] arr){
        String[] board = {"WBWBWBWB", "BWBWBWBW"}; // 화이트로 시작하는 고정 배열

        // 화이트를 기준으로 최소 비용 계산
        int whiteCount = 0;
        for (int i = 0; i < 8; i++) {
            int row = startRow + i;
            for (int j = 0; j < 8; j++) {
                int col = startCol + j;
                if (arr[row].charAt(col) != board[row % 2].charAt(j)){
                    whiteCount++;
                }
            }
        }
        // blackCount(블랙버전 체스판 최소 비용) = 64(8*8) - whiteCount
        return Math.min(whiteCount, 64 - whiteCount);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        // ** 최소 칠해야하는 Count
        // 현재상태, B 체스판 최소비용, W 체스판 최소비용 비교

        // 1. NxM 배열 생성
        // - 크기는 8x8 최소로 크기를 잘라야함
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int m = Integer.parseInt(input[1]);

        String[] arr = new String[n];
        for (int i = 0; i < n; i++) {
            arr[i] = br.readLine();
        }

        //2. 체스판 자르기
        int min = Integer.MAX_VALUE;
        // n-8 = 8x8로 만들어주기 위한 최소범위
        // ex) if n = 10, i = 10-8 = 2
        // -> (0,0) / (1,0) / (2,0) 부터 시작하는 범위 탐색
        for (int i = 0; i <= n-8; i++) {
            for (int j = 0; j <= m-8; j++) {
                // 3. 현재 체스판의 최소 비용
                int curMin = getSolution(i, j, arr);
                // 4. 전체 최소 값 비교 후 갱신
                if (min > curMin)
                    min = curMin;
            }
        }

        bw.write(min + "");
        bw.flush();
        bw.close();
    }
}

File: 이하영/240517/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
n, m = map(int, input().split())
board = []
result = []
for _ in range(n):
    board.append(input())

for x in range(n-7):
  for y in range(m-7): #8*8크기 고정 후 탐색
    black_paint = 0 
    white_paint = 0 
    for i in range(x, x+8):
      for j in range(y, y+8):
        if(i+j) % 2 == 0:   #짝수끼리 색이 같음
          if board[i][j] != 'W': #첫 시작이 흰색이 아닐 때
            white_paint += 1
          if board[i][j] != 'B': #첫 시작이 검정이 아닐 때
            black_paint += 1
        else:               #홀수끼리 색이 같
          if board[i][j] != 'B': 
            white_paint += 1
          if board[i][j] != 'W':
            black_paint += 1 

    result.append(white_paint)
    result.append(black_paint)


print(min(result))